### ISSUE TITLE
could not able to test webrtc

### ISSUE THREAD

**Issue by imranshaaz** [AirenSoft/OvenMediaEngine#980]
Before asking for help, please read the troubleshooting section in the manual carefully.
https://airensoft.gitbook.io/ovenmediaengine/troubleshooting

Please fill out the form below to create an issue. Failure to follow the form may result in late responses or close issues.

**PROBLEM**

I did able to setup ovenmedia engine on ubuntu20 on a GCP vm, I connect and able to start the streaming using OBS, but could not able to test it over ovenmedia test player using ws://xxxx:3333/app/stream.
When I hit the url in test play it shows "connection with low-latency server failed. 

And I am not able to connect over webrtc url with the details server.xml file provided(I just updated the server's external IP for Origin server block)

**EXPECTATION**

I would to do the webrtc connection/streaming successful for testing. Once it is successful I will do the production setup. 

**ENVIRONMENTAL INFORMATION**

- OvenMediaEngine Information

Please check whether it is reproduced in the master branch first. The Docker image for the master branch is airensoft/ovenmediaengine:dev .
If you are using the Master branch, please write Master.
If you are using Release, please write the version (eg 0.12.4).
If you are using Docker by building it yourself, please tell us what version of the image you built.
If you are using a public Docker image, please provide the image path with tags. (e.g. airensoft/ovenmediaengine:0.12.8)

- Encoder information

OBS 28.1.2 And how is the codec set on that encoder?
If you are using OvenLiveKit as an encoder, please let me know if it reproduces here 
Using OvenLiveKit I am able to do the strming, via rtmp connected. 
https://demo.ovenplayer.com/demo_input.html. And please tell me what error is output in the browser's developer console.
Error: Failed to construct 'WebSocket': An insecure WebSocket connection may not be initiated from a page loaded over HTTPS.
If you are using RTSP Pull, please let me know which RTSP server you are using.

- Player information

Please let me know if it reproduces on http://demo.ovenplayer.com or https://demo.ovenplayer.com. And please tell me what error is output in the browser's developer console.

Err: Connection with low-latency server failed. 
- URL: http://demo.ovenplayer.com/#%7B%22playerOption%22%3A%7B%22autoStart%22%3Atrue%2C%22autoFallback%22%3Atrue%2C%22mute%22%3Afalse%2C%22sources%22%3A%5B%7B%22type%22%3A%22webrtc%22%2C%22file%22%3A%22ws%3A%2F%2F34.76.248.187%3A3333%2Fapp%2Fstream%22%7D%5D%2C%22webrtcConfig%22%3A%7B%22timeoutMaxRetry%22%3A4%2C%22connectionTimeout%22%3A10000%2C%22playoutDelayHint%22%3A3%7D%2C%22hlsConfig%22%3A%7B%22liveSyncDuration%22%3A1.5%2C%22targetLatency%22%3A3%2C%22maxLiveSyncPlaybackRate%22%3A1.1%7D%7D%2C%22demoOption%22%3A%7B%22autoReload%22%3Atrue%2C%22autoReloadInterval%22%3A2000%7D%7D
- 

Please tell me which address you used to send and play. You can hide IP or Domain.
rtmp://xxxx:1935/app
ws://xxxx:3333/app/stream

- Operating system and version

OS and version (e.g. Ubuntu 20.04)

**SETUP INFORMATION AND LOGS**

Please upload Server.xml.
Please upload the entire file /var/log/ovenmediaengine/ovenmediaengine.log. (You may delete your personal information.)
It's hard to find the problem if you post only part of the log. Be sure to upload the entire file.

**OTHER HELPFUL INFORMATION**

Please write any information that anyone viewing this issue can reference.
[testlogs.txt](https://github.com/AirenSoft/OvenMediaEngine/files/10254205/testlogs.txt)

**Comment by naanlizard** [comment]
Have you opened the appropriate ports in your firewall?

**Comment by imranshaaz** [comment]
Yes, I have opened all the ports for public(0.0.0.0/0). I am sorry my
questions are very silly.
Honestly I am trying to setup this env since 2weeks, but could not get it
done. Is required to install the TURN servers separately? And should I give
the TCP relay server's ip same as vm's external IP or TURN server IP?
Appreciate your help to make it.
 Thank you.

**Comment by naanlizard** [comment]
For testing it is easiest to leave the default google TURN server

What is your OBS stream and streamkey set to?

**Comment by imranshaaz** [comment]
Stream I set in OBS rtmp://xxxx:1935/app
And the key: stream

**Comment by naanlizard** [comment]
Ah, I see at least one problem (apologies, your ticket is messy)

You are trying to load an insecure websocket on a secure page. Either use the http (not HTTPS) demo player or connect to the secure webrtc port instead

**Comment by imranshaaz** [comment]
I am using http only, ws://xxxx:3333/app/stream. I haven't configured TLS,
that is why I am trying with ws.

**Comment by getroot** [comment]
![image](https://user-images.githubusercontent.com/17779085/208331466-c4601c0e-1f2f-445d-8bc1-90c518184f1d.png)

I put the rtmp input into your server and tried to play it and it seems to work.

**Comment by imranshaaz** [comment]
Thanks,
For me it is not working, I am getting the error, attached the screenshot.

**Comment by getroot** [comment]
Your screenshot is not attached. Please upload again.
Have you tried loading ws:// from https://demo.ovenplayer.com? Then there is the mixed content issue. Try it at http://demo.ovenplayer.com.

**Comment by imranshaaz** [comment]
Yes, I tried with http://demo.ovenplayer.com only. 

<img width="960" alt="ws-error" src="https://user-images.githubusercontent.com/49882022/208386924-998b763b-0af3-4354-99de-dd9e8d235bb0.png">

**Comment by umeshshende** [comment]
@imranshaaz did you add transport TCP to the string while publishing i was also facing same problem but when i add transport TCP for ws connection it works
